### PR TITLE
announcement models

### PR DESCRIPTION
**Pull request by wangqiang1208**
Announcement models, with basic attributes.

### CODE REVIEW

File: app/models/announcement.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Announcement < ActiveRecord::Base
```
Comment (lowjoel) [inline app/models/announcement.rb]: `Course::Announcement`

File: db/migrate/20150112133542_create_announcements.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class CreateAnnouncements < ActiveRecord::Migration
+  def change
+    create_table :announcements do |t|
```
Comment (lowjoel) [inline db/migrate/20150112133542_create_announcements.rb]: `course_announcements`. we have `instance_announcements` and `announcements` (global)

File: db/migrate/20150114024350_create_course_announcements.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class CreateCourseAnnouncements < ActiveRecord::Migration
+  def change
+    create_table :course_announcements do |t|
+      t.references :course, null: false
+      t.string :title, null: false
+      t.text :content
```
Comment (lowjoel) [inline db/migrate/20150114024350_create_course_announcements.rb]: is `content` better or `description` better?
@wangqiang1208 @jsyeo @weiqingtoh

File: db/migrate/20150114024350_create_course_announcements.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+      t.datetime :publish_at, null: false
```
Comment (lowjoel) [inline db/migrate/20150114024350_create_course_announcements.rb]: I wrote the `t.time_bounded` helper for this.

File: db/migrate/20150114024350_create_course_announcements.rb (hunk 3 above)
Comment (wangqiang1208) [inline db/migrate/20150114024350_create_course_announcements.rb]: I prefer content, description is like describe something.  So I use
description for course model

Comment (weiqingtoh) [inline db/migrate/20150114024350_create_course_announcements.rb]: Comparing between `announcement.content` vs `announcement.description` tells me that either works. If there must be a choice, I agree with @wangqiang1208  that content is more accurate, while the description might be used to describe the Announcement

**Pull request merged**